draft-ietf-tls-external-psk-importer.md

```diff
@@ -88,6 +88,39 @@ a tuple of (Base Key, External Identity, KDF). The associated KDF (and hash func
 - External Identity: The identity of an EPSK.
 - Imported Identity: The identity of a PSK as sent on the wire.
 
+# Binder Key
+To prevent PSK Importers from being confused with standard OOB PSKs we change the label used in the computation of the PSK binder key.
+In TLS 1.3 the PSK binder key computation is defined as follows:
+
+~~~
+             0
+             |
+             v
+   PSK ->  HKDF-Extract = Early Secret
+             |
+             +-----> Derive-Secret(., "ext binder" | "res binder", "")
+             |                     = binder_key
+~~~
+
+We replace the string "ext binder" with "imp binder".
+This means the binder key is now computed as follows:
+
+~~~
+             0
+             |
+             v
+   PSK ->  HKDF-Extract = Early Secret
+             |
+             +-----> Derive-Secret(., "ext binder"
+             |                      | "res binder"
+             |                      | "imp binder", "")
+             |                     = binder_key
+~~~
+
+The only change that is not a previously valid TLS 1.3 trace in the change to the `binder_key` computation.
```

Comment:
I don't see this as a change, but more of an addition.

Reply:
> This seems harmless, but it is predicated on a need to separate keys that I don't understand. Is there more rationale somewhere?

The underlying issue is that without this change you can't distinguish the plain OOB case from the PSK Importer case. 
For example the server could believe it was using an importer PSK, but the client could believe it was using a vanilla OOB key. Because there is no extension in the handshake signaling the use of Importer PSKs explicitly we need to agree the use of Importer keys implicitly. 

I actually think this a weakness of TLS 1.3's OOB keys in general. They should have had a way to agree security context around the establishment of PSKs, in the same way that exporter keys allow us to bind a TLS context into a higher layer protocol. 
This change, in conjunction with PR #12 resolves this issue. 
Even if you do not want to change the hash function you can use PSK importers to bind in extra security context, and the key splitting will ensure everything Just Works™ (i.e. prove that the client and server agree on the extra security context).

Reply:
@jhoyla I think @martinthomson is asking to add rationale for the new label. Perhaps you could summarize this in a single sentence in the PR?